bitnami/pinniped/templates/supervisor/service-api.yaml

```
@@ -0,0 +1,60 @@
+{{- if .Values.supervisor.enabled }}
+apiVersion: v1
+kind: Service
+metadata:
+  name: {{ template "pinniped.supervisor.api.fullname" . }}
+  namespace: {{ include "common.names.namespace" . | quote }}
+  labels: {{- include "common.labels.standard" . | nindent 4 }}
+    app.kubernetes.io/part-of: pinniped
+    app.kubernetes.io/component: supervisor
+    {{- if .Values.supervisor.service.labels }}
+    {{- include "common.tplvalues.render" ( dict "value" .Values.supervisor.service.labels "context" $ ) | nindent 4 }}
+    {{- end }}
+    {{- if .Values.commonLabels }}
+    {{- include "common.tplvalues.render" ( dict "value" .Values.commonLabels "context" $ ) | nindent 4 }}
+    {{- end }}
+  {{- if or .Values.supervisor.service.annotations .Values.commonAnnotations }}
+  annotations:
+    {{- if .Values.supervisor.service.annotations }}
+    {{- include "common.tplvalues.render" ( dict "value" .Values.supervisor.service.annotations "context" $) | nindent 4 }}
+    {{- end }}
+    {{- if .Values.commonAnnotations }}
+    {{- include "common.tplvalues.render" ( dict "value" .Values.commonAnnotations "context" $ ) | nindent 4 }}
+    {{- end }}
+  {{- end }}
+spec:
+  type: {{ .Values.supervisor.service.type }}
+  {{- if and .Values.supervisor.service.clusterIP (eq .Values.supervisor.service.type "ClusterIP") }}
+  clusterIP: {{ .Values.supervisor.service.clusterIP }}
+  {{- end }}
+  {{- if .Values.supervisor.service.sessionAffinity }}
+  sessionAffinity: {{ .Values.supervisor.service.sessionAffinity }}
+  {{- end }}
+  {{- if .Values.supervisor.service.sessionAffinityConfig }}
+  sessionAffinityConfig: {{- include "common.tplvalues.render" (dict "value" .Values.supervisor.service.sessionAffinityConfig "context" $) | nindent 4 }}
+  {{- end }}
+  {{- if or (eq .Values.supervisor.service.type "LoadBalancer") (eq .Values.supervisor.service.type "NodePort") }}
+  externalTrafficPolicy: {{ .Values.supervisor.service.externalTrafficPolicy | quote }}
+  {{- end }}
+  {{- if and (eq .Values.supervisor.service.type "LoadBalancer") (not (empty .Values.supervisor.service.loadBalancerSourceRanges)) }}
+  loadBalancerSourceRanges: {{ .Values.supervisor.service.loadBalancerSourceRanges }}
+  {{- end }}
+  {{- if and (eq .Values.supervisor.service.type "LoadBalancer") (not (empty .Values.supervisor.service.loadBalancerIP)) }}
+  loadBalancerIP: {{ .Values.supervisor.service.loadBalancerIP }}
+  {{- end }}
+  ports:
+    - name: https
+      port: {{ .Values.supervisor.service.ports.https }}
+      protocol: TCP
+      targetPort: {{ .Values.supervisor.service.ports.aggregatedapiserver }}
```

If you accept my suggestion below you have to change this line also

```suggestion
      targetPort: {{ .Values.supervisor.service.ports.aggregatedAPIServer }}
```